### PR TITLE
Add DNS import via nameserver & improve DNS importing

### PR DESCRIPTION
**Pull request by ad-m**

### CODE REVIEW

**Comment by ad-m** [comment]
@fredericosilva , resubmit.

**Comment by fredericosilva** [comment]
@ad-m is this still valid?

File: bin/dns/recordTypes.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ module.exports = {
         to_bind: content => ({
             alias: content,
         }),
-        to_content: record => record.alias,
+        to_content: (record, name) => formatRecordName(record.alias, name),
```
Comment (ad-m) [inline bin/dns/recordTypes.js]: Our API always expects FQDN, so we need to convert.

File: bin/dns/recordTypes.js [hunk 2]
```diff
@@ -46,14 +46,14 @@ module.exports = {
             preference: content.split(' ')[0],
             host: content.split(' ')[1],
         }),
-        to_content: record => `${record.preference} ${record.host}`,
+        to_content: (record, name) => `${record.preference} ${formatRecordName(record.host, name)}`,
```
Comment (ad-m) [inline bin/dns/recordTypes.js]: Our API always expects FQDN, so we need to convert.

File: bin/dns/recordTypes.js [hunk 3]
```diff
@@ -46,14 +46,14 @@ module.exports = {
...
     },
     ns: {
         value: 'ns1.example.com',
         to_bind: content => ({
             host: content,
         }),
-        to_content: (record, zone) => formatRecordName(record.host, zone.dnsName),
+        to_content: (record, name) => formatRecordName(record.host, name),
```
Comment (ad-m) [inline bin/dns/recordTypes.js]: Our API always expects FQDN, so we need to convert.

File: bin/dns/recordTypes.js [hunk 4]
```diff
@@ -63,7 +63,7 @@ module.exports = {
             port: content.split(' ')[2],
             target: content.split(' ')[3],
         }),
-        to_content: record => `${record.priority} ${record.weight} ${record.port} ${record.target}`,
+        to_content: (record, name) => `${record.priority} ${record.weight} ${record.port} ${formatRecordName(record.target, name)}`,
```
Comment (ad-m) [inline bin/dns/recordTypes.js]: Our API always expects FQDN, so we need to convert.

File: bin/dns/tests.js [hunk 5]
```diff
@@ -301,6 +301,15 @@ ava.serial('dns punycode encoded', async t => {
     await tests.remove('dns zone', zone);
 });
 
+ava.serial('dns create with probing', async t => {
+    const value = '3.3.3.3';
+    const name = `${tests.getName(t.title)}.${value}.xip.io.`;
+    const zone = await tests.run(`dns zone create --type public --name ${name} --dns-probing`);
```
Comment (ad-m) [inline bin/dns/tests.js]: Initial reason for developing PR.

File: bin/dns/zone/import/index.js [hunk 6]
```diff
@@ -33,70 +33,92 @@ const supported_label = supported_types
     .map(x => x.toUpperCase())
     .join(', ');
 
+const equalRrset = (a, b) => a.record.length == b.record.length &&
+    a.record.every(x => b.record.some(y => y.content == x.content)) &&
+    b.record.every(x => a.record.some(y => y.content == x.content));
+
+const zonefile2rrset = (fname, dnsName) => {
```
Comment (ad-m) [inline bin/dns/zone/import/index.js]: The extracted logic of converting the zonefile structure from the library to our own structure. I have developed it and would like to keep it, although the initial cause of the change is no longer present (approach changed).

**Comment by ad-m** [comment]
> @ad-m is this still valid?

The proposed changes to this PR are still valid, but I see that the initial cause of PR is limited.

**Pull request merged**